x/mongo/driver/topology/topology_test.go

```diff
@@ -230,6 +232,29 @@ func TestServerSelection(t *testing.T) {
 			t.Fatalf("did not receive error from server selection")
 		}
 	})
+	t.Run("Context Deadline Error", func(t *testing.T) {
+		desc := description.Topology{}
+		topo, err := New()
+		noerr(t, err)
+		subCh := make(chan description.Topology, 1)
+		subCh <- desc
+		resp := make(chan error)
```

Comment:
nit: You don't need a channel anymore. This was needed when the code below had to wait until an error or a timeout, which required selecting over two channels, but that part is now handled by `assert.Soon`. This can be:

```
var serverSelectionError error
callback := func() {
    // setup topology and server selection state
    // setup ctx
    _, serverSelectionError = topo.selectServerFromSubscription(ctx, subCh, state)
}

assert.Soon(callback, 100*time.Millisecond)
// assertions about serverSelectionError being correct
```

Note the use of `=` rather than `:=` in the callback to avoid shadowing.